runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -193,7 +193,7 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
         for (int x = 0; x < rt.getParameters().getTrimRetry(); x++) {
             // Linearize this read against a timestamp
             final long timestamp = rt.getSequencerView()
-                            .query(getStreamID()).getToken().getSequence();
+                            .query(getStreamID()).getStreamTails().get(0);
```

Comment:
Maybe hide this implementation in the SequencerView -- leaking the list "de-referencing  logic" doesn't seem clean.

Reply:
+1

Reply:
Done.

Reply:
Done.